src/lint/lint.nim

```diff
@@ -1,5 +1,55 @@
-proc lint* =
-  echo "No linting rules have yet been implemented.\n" &
-       "As we're in the process of implementing the linting rules, please " &
-       "re-run this command regularly to see if your track passes the " &
-       "latest linting rules."
+import std/[json, os, strformat, terminal]
+import ".."/[cli, helpers]
+
+template writeError(description: string, details: string) =
+  stdout.styledWriteLine(fgRed, description & ":")
+  stdout.writeLine(details)
+  stdout.write "\n"
+  result = false
+
+proc isValidTrackConfig(trackDir: string): bool =
+  result = true
+  let configJsonPath = trackDir / "config.json"
+  if fileExists(configJsonPath):
+    try:
+      let j = parseFile(configJsonPath)
+    except:
+      writeError("JSON parsing error", getCurrentExceptionMsg())
+  else:
+    writeError("Missing file", configJsonPath)
+
+proc conceptExerciseFilesExist(trackDir: string): bool =
+  const
+    conceptExerciseFiles = [
+      ".docs" / "hints.md",
+      ".docs" / "instructions.md",
+      ".docs" / "introduction.md",
+      ".meta" / "config.json",
+    ]
+
+  let conceptDir = trackDir / "exercises" / "concept"
+  result = true
+
+  if dirExists(conceptDir):
+    for dir in getSortedSubDirs(conceptDir):
+      for conceptExerciseFile in conceptExerciseFiles:
+        let path = dir / conceptExerciseFile
+        if not fileExists(path):
+          writeError("Missing file", path)
```

Comment:
We have to think whether we want to check concept exercises based on the directories, or based on the config.json file's contents. But this is fine for now